### PR TITLE
Add an uninitialized stack variable mission.

### PR DESCRIPTION
**Pull request by bsdjhb**

### CODE REVIEW

File: src/missions/uninitialized-stack-frame-control-flow/README.md [hunk 1]
```diff
@@ -1,3 +1,44 @@
...
+provided as a pair of hexadecimal characters (case is ignored).  Each
+cookie is stored at successive bytes in an on-stack character array.
+The character array aliases an uninitialized function pointer used in
+a subsequent function.  A minus character ('-') can be used to skip
+over a character in the array without providing a new cookie.
+Whitespace is ignored in the input file.
+
+### Building and running
+
+All versions of the test program must be built with `-O0`.  The
```
Comment (jrtc27) [inline src/missions/uninitialized-stack-frame-control-flow/README.md:34]: You could instead make the two `cookie_fn`s volatile.

File: src/missions/uninitialized-stack-frame-control-flow/README.md [hunk 2]
```diff
@@ -1,3 +1,44 @@
...
+hardened CHERI-RISC-V version with stack clearing is built by adding
+`-ftrivial-auto-var-init=zero
+-enable-trivial-auto-var-init-zero-knowing-it-will-be-removed-from-clang`.
```
Comment (jrtc27) [inline src/missions/uninitialized-stack-frame-control-flow/README.md:37]: This cracks me up every time..........

File: src/missions/uninitialized-stack-frame-control-flow/README.md [hunk 3]
```diff
@@ -1,3 +1,44 @@
...
+   exploitable without any CHERI protections.
+
+2. A hardened CHERI-RISC-V compilation with stack clearing, which
+   should be non-exploitable.
+
+3. A baseline CHERI-RISC-V compilation with no stack clearing, which
+   should be non-exploitable due to pointer tagging.
+
+The success condition for an exploit, given attacker-provided input
+overriding an on-stack bufer, is to modify control flow in the program
```
Comment (jrtc27) [inline src/missions/uninitialized-stack-frame-control-flow/README.md:18]:
```suggestion
overriding an on-stack buffer, is to modify control flow in the program
```

File: src/missions/uninitialized-stack-frame-control-flow/README.md [hunk 4]
```diff
@@ -1,3 +1,44 @@
...
+of the program:
+
+1. A baseline RISC-V compilation, to establish that the vulnerability is
+   exploitable without any CHERI protections.
+
+2. A hardened CHERI-RISC-V compilation with stack clearing, which
+   should be non-exploitable.
+
+3. A baseline CHERI-RISC-V compilation with no stack clearing, which
+   should be non-exploitable due to pointer tagging.
```
Comment (jrtc27) [inline src/missions/uninitialized-stack-frame-control-flow/README.md:15]: It's exploitable when the input matches '^-*$`, no?

File: src/missions/uninitialized-stack-frame-control-flow/README.md (hunk 2 above)
Comment (brooksdavis) [reply src/missions/uninitialized-stack-frame-control-flow/README.md:37]: I wonder if we should link to the LLVM mailing list thread on the subject.

File: src/missions/uninitialized-stack-frame-control-flow/README.md (hunk 4 above)
Comment (bsdjhb) [reply src/missions/uninitialized-stack-frame-control-flow/README.md:15]: I should perhaps rename `nothing_doing` to `no_cookies` to be more inline with the theme.

**Comment by brooksdavis** [comment]
I think usability would benefit from being able to terminate input with `.` or the like. I also found it briefly confounding that IO was buffered and thus I had to either press Ctrl-D twice or Enter and then Ctrl-D.

**Comment by bsdjhb** [comment]
> I think usability would benefit from being able to terminate input with `.` or the like. I also found it briefly confounding that IO was buffered and thus I had to either press Ctrl-D twice or Enter and then Ctrl-D.

Hmm, or even just the first newline perhaps.  I had ignored whitespace to follow what the temporal mission did, but this needs less fancy input.

**Comment by bsdjhb** [comment]
I've done the first newline as a terminator.  I will now work on trying to get this working with both -O2 and -O0 and then push the result as a second commit so we can compare the two versions to see which we want.  However, I wanted to get the other cleanups pushed so the diff was clear for -O2.

**Comment by jrtc27** [comment]
If it works then this seems fine now from my perspective.

**Pull request merged**